chainercv/datasets/transform_dataset.py

```diff
@@ -38,6 +38,9 @@ def __init__(self, dataset, transform):
         self._dataset = dataset
         self._transform = transform
 
+    def __getattr__(self, key):
+        return getattr(self._dataset, key)
+
```

Comment:
Thanks.
This is a nice functionality to add, but it will not work together with other dataset functions supported by Chainer.

For example, `chainer.datasets.SubDataset` does not have the functionality you added. Therefore, If used together with `chainer.datasets.SubDataset`, the user will no longer be able to access functions supported by the underlying dataset.

I think this is an issue that needs to be discussed on the main Chainer repository.